app/src/main/java/com/kickstarter/viewmodels/projectpage/AddOnsViewModel.kt

```diff
@@ -0,0 +1,207 @@
+package com.kickstarter.viewmodels.projectpage
+
+import android.util.Pair
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.kickstarter.libs.Environment
+import com.kickstarter.libs.rx.transformers.Transformers
+import com.kickstarter.libs.utils.RewardUtils
+import com.kickstarter.libs.utils.extensions.addToDisposable
+import com.kickstarter.mock.factories.ShippingRuleFactory
+import com.kickstarter.models.Reward
+import com.kickstarter.models.ShippingRule
+import com.kickstarter.ui.data.ProjectData
+import io.reactivex.Observable
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.PublishSubject
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+
+data class AddOnsUIState(
+    var currentShippingRule: ShippingRule = ShippingRule.builder().build(),
+    var shippingSelectorIsGone: Boolean = false,
+    var currentAddOnsSelection: MutableMap<Reward, Int> = mutableMapOf()
+)
+class AddOnsViewModel(val environment: Environment) : ViewModel() {
+    private val disposables = CompositeDisposable()
+    private val currentConfig = requireNotNull(environment.currentConfigV2())
+
+    private val currentUserReward = PublishSubject.create<Reward>()
+    private val shippingRules = BehaviorSubject.create<List<ShippingRule>>()
+    private val defaultShippingRule = PublishSubject.create<ShippingRule>()
+    private val shippingRuleSelected = PublishSubject.create<ShippingRule>()
```

Comment:
If possible we would want to move away from RX as much as possible (some of our more established classes that use observables i understand the need to use).  In most cases just having a variable we can change and then emitting changes based on those variables can work.  

When a user selects a new reward (current user reward)instead of emitting an observable we can just make the calls as needed int he viewmodel and emit the UI state, etc.

For the sake of time we can keep this for now, but for future projects we want as little RX as possible added in.